goss-testing/scripts/postgres_replication_lag.sh

```diff
@@ -1,43 +1,119 @@
 #!/bin/bash
 
 print_results=0
-while getopts ph stack
+exit_on_failure=0
+while getopts pem:a:w:h stack
 do
     case "${stack}" in
           p) print_results=1;;
-          h) echo "usage: postgres_replication_lag.sh           # Only print 'PASS' upon success"
+          e) exit_on_failure=1;;
+          m) POSTGRES_MAX_LAG=$OPTARG;;
+          a) POSTGRES_MAX_ATTEMPTS=$OPTARG;;
+          w) POSTGRES_WAIT_SECONDS_BETWEEN_ATTEMPTS=$OPTARG;;
+          h) echo "usage: postgres_replication_lag.sh           # Print 'PASS' upon success"
              echo "       postgres_replication_lag.sh -p        # Print all results and errors if found. Use for manual check."
-             exit 3;;
-         \?) echo "usage: postgres_replication_lag.sh           # Only print 'PASS' upon success"
+             echo "       postgres_replication_lag.sh -p -e     # Print all results and errors if found. Exit if failure is encountered."
+             echo "       postgres_replication_lag.sh -p -m <max_allowed_lag> -a <number_attempts> -w <wait_seconds_between_attempts>"
+             echo "default parameters: -m 0  -a 10  -w 10"
+             exit 4;;
+         \?) echo "usage: postgres_replication_lag.sh           # Print 'PASS' upon success"
              echo "       postgres_replication_lag.sh -p        # Print all results and errors if found. Use for manual check."
-             exit 3;;
+             echo "       postgres_replication_lag.sh -p -e     # Print all results and errors if found. Exit if failure is encountered."
+             echo "       postgres_replication_lag.sh -p -m <max_allowed_lag> -a <number_attempts> -w <wait_seconds_between_attempts>"
+             echo "default parameters: -m 0  -a 10  -w 10"
+             exit 4;;
     esac
 done
 
+# The POSTGRES_MAX_LAG environment variable may be exported by the user to control
+# the maximum lag value permitted by this script. Setting its value to
+# a negative number or a non-integer value has the effect of skipping the
+# maximum lag check. In other words, if one wishes to skip this check,
+# one could:
+# export POSTGRES_MAX_LAG=skip
+POSTGRES_MAX_LAG=${POSTGRES_MAX_LAG:-'0'}
+
+# POSTGRES_MAX_ATTEMPTS specifies the maximum number of times the
+# PostgreSQL check will be performed on a given cluster before failing. Note that
+# failures other than due to maximum lag are always fatal and are not retried.
+# If unset or set to a non-positive integer, default to 10
+if [[ ! $POSTGRES_MAX_ATTEMPTS =~ ^[0-9][0-9]*$ ]]; then
```

Comment:
this regular expression matches 0, so I don't think this does what the documentation above says since 0 is a non-positive integer.

Reply:
I changed this to` [[ ! $POSTGRES_MAX_ATTEMPTS =~ ^[1-9][0-9]*$ ]]`. We do not want 0 to be a valid input because the test will not run if the input is 0. I also think it is okay to assume the number entered will not start with a 0.